### PR TITLE
#184: Bank account choice lost when page is refreshed

### PR DESCRIPTION
**Pull request by romanstrobl**
This PR contains all changes from #186 due to merge conflicts, so it is built on top of #186.

This change resolves a case when user refreshes page (or switches language) and bank account choice is lost because operation is retrieved from the backend which doesn't know the bank account choice yet.

I also fixed response objects in methods which are not related to authentication and thus should not return next steps.

### CODE REVIEW

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java [hunk 1]
```diff
@@ -165,7 +154,9 @@ protected UpdateOperationResponse authorize(String operationId, String userId) t
      * @return Response with information about operation update result.
      * @throws NextStepServiceException In case communication fails.
      */
-    protected UpdateOperationResponse authorize(String operationId, String userId, List<KeyValueParameter> params) throws NextStepServiceException {
+    protected UpdateOperationResponse authorize(String operationId, String userId, List<KeyValueParameter> params) throws NextStepServiceException, AuthStepException {
+        // get operation to validate operation state
+        getOperation(operationId);
```
Comment (petrdvorak) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: Are we just using a side effect here? Can we just validate the operation?

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java [hunk 2]
```diff
@@ -188,7 +179,9 @@ protected UpdateOperationResponse authorize(String operationId, String userId, L
      * @return Response with information about operation update result.
      * @throws NextStepServiceException In case communication fails.
      */
-    protected UpdateOperationResponse failAuthorization(String operationId, String userId, List<KeyValueParameter> params) throws NextStepServiceException {
+    protected UpdateOperationResponse failAuthorization(String operationId, String userId, List<KeyValueParameter> params) throws NextStepServiceException, AuthStepException {
+        // get operation to validate operation state
+        getOperation(operationId);
```
Comment (petrdvorak) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: dtto.

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java [hunk 3]
```diff
@@ -210,7 +203,9 @@ protected UpdateOperationResponse failAuthorization(String operationId, String u
      * @return Response with information about operation update result.
      * @throws NextStepServiceException In case communication fails.
      */
-    protected UpdateOperationResponse cancelAuthorization(String operationId, String userId, OperationCancelReason cancelReason, List<KeyValueParameter> params) throws NextStepServiceException {
+    protected UpdateOperationResponse cancelAuthorization(String operationId, String userId, OperationCancelReason cancelReason, List<KeyValueParameter> params) throws NextStepServiceException, AuthStepException {
+        // get operation to validate operation state
+        getOperation(operationId);
```
Comment (petrdvorak) [inline powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: dtto

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: Fixed.

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java (hunk 2 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: Fixed.

File: powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java (hunk 3 above)
Comment (romanstrobl) [reply powerauth-webflow-authentication/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/controller/AuthMethodController.java]: Fixed.

**Pull request merged**